docs/source/basics/prompt_templates.rst

```diff
@@ -0,0 +1,249 @@
+.. _prompt_templates_usage_label:
+
+================
+Prompt Templates
+================
+
+Prompt templates are structured text templates which are used to format user prompts
+to optimize model performance on specific tasks. They can serve many purposes:
```

Comment:
Again, lead with an example. I imagine the prompt templates page will get a lot of eyes, just give a very basic example of "here is a question I want to ask an LLM, here is how it can be formatted" (maybe with some distinction between a training case vs an inference case)

Reply:
maybe I can just take GrammarCorrectionTemplate as an example task and show how it's used.